### DISCUSSION TITLE
Full export & import

### DISCUSSION THREAD

**Discussion by lnwara** [keycloak/keycloak#25465] [Archive]
We had create a keycloak instance (quay.io/keycloak/keycloak:21.0.1) for development and apply all the necessary users, groups, roles and localizations with message bundles, all works fine. now we need to move to the next environment with full export but there was only the partial export in the admin console, so is there and workaround or command that may help on full export. since we will make this process automated for each environment.

**Comment by robson90** [comment]
Have a look here: https://www.keycloak.org/server/importExport

**Reply by flythebluesky** [reply]
I think the OP is saying that it would be nice to be able to do a full realm export against a running Keycloak instance. This would be useful for development, testing, migration to other instances etc. I imagine the realm would need to be temporarily disabled to ensure consistency.
